juju/unit.py

```diff
@@ -108,21 +109,58 @@ def resolved(self, retry=False):
         :param str command: The command to run
         :param int timeout: Time to wait before command is considered failed
 
+        Returns a tuple containing the stdout, stderr, and return code
+        from the command.
+
         """
         action = client.ActionFacade()
         action.connect(self.connection)
 
         log.debug(
             'Running `%s` on %s', command, self.name)
 
-        # TODO this should return an Action
-        return await action.Run(
+        action_status = asyncio.Queue(loop=self.model.loop)
```

Comment:
Aha. I missed that I could pass the tag to _wait_for_new as the entity_id. That makes a lot of sense, and definitely saves on all my boilerplate. Making the change locally now. Thx :-)